library/src/main/java/com/google/android/exoplayer2/extractor/mp3/Mp3Extractor.java

```diff
@@ -329,7 +361,8 @@ private Seeker setupSeeker(ExtractorInput input) throws IOException, Interrupted
       frame.setPosition(xingBase);
       headerData = frame.readInt();
     }
-    if (headerData == XING_HEADER || headerData == INFO_HEADER) {
+    if ((flags & FLAG_ENABLE_CONSTANT_BITRATE_SEEKING) == 0
```

Comment:
I don't think it's right that you're preventing the creation of the XingSeeker in all cases when the flag is set. If there's a Xing header that *does* have a table of contents, it's going to be better to use the XingSeeker than to force constant bitrate seeking...

It seems more appropriate to leave this code unchanged and instead modify the
```
if(seeker == null) {
  ...
}
```
block at the bottom of this method to be more like:
```
if (seeker == null || (!seeker.isSeekable()
    && (flags & FLAG_ENABLE_CONSTANT_BITRATE_SEEKING) != 0))) {
  ...
}
```